mesonbuild/interpreter.py

```diff
@@ -3088,6 +3088,8 @@ def func_run_target(self, node, args, kwargs):
             if not isinstance(i, (str, build.BuildTarget, build.CustomTarget, dependencies.ExternalProgram, mesonlib.File)):
                 mlog.debug('Wrong type:', str(i))
                 raise InterpreterException('Invalid argument to run_target.')
+            if isinstance(i, dependencies.ExternalProgram) and not i.found():
+                raise InterpreterException('Tried to use non-existing executable %s.' % i.name)
```

Comment:
Two things:

1. We should never end our error messages with a full stop, especially when printing some information because then the user will wonder whether `fooprog` wasn't found or `fooprog.`

2. We should use `{!r}` instead of `%s` because it will give a printable representation in quotes, which will cover for instance weird unprintable characters, not-string objects, etc.

Reply:
1. I considered having `%s` in double quotes to show the name clearly separated. Would that work?

I have to admit I have trouble parsing the sentence. Do you mean that someone has searched for an executable called `fooprog.` (with the period in the file name)? That seems very unlikely, I don't remember ever seeing a command whose name ends in a period. And regardless the search results have already been printed above so people should be able to see what is the actual thing that failed.

Reply:
We have previously had people complain that having a period at the end of the sentence when something is being printed confuses them as to whether the period was a part of the thing being printed. No valid command would have an `.` at the end, but invalid ones might, or it might be a bug, the user can't know.

Reply:
> I considered having %s in double quotes to show the name clearly separated. Would that work?

That might work, but `{!r}` is more bullet proof, because if an object without `__str__()` is passed to `'%s' % foo` Python will print a traceback instead.